### ISSUE TITLE
Telegram stoploss alert feature

### ISSUE THREAD

**Issue by MrHumanRebel** [freqtrade/freqtrade#4579]
Hi! 

I have been thinking about a telegram feature, I don't really have that much experience in coding so I don't know if this is possible or not. 
The basic idea: the telegram bot would send a message at a given minus percentage that the current investment is about to reach stoploss. For example the stoploss is -0.5 and the stoploss alert rate is 90% of the given stoploss, then send a message at -0.45. The main advantage would be that the admin could review and decide to keep the bot running and potentially reach stoploss and sell or stop the bot temporarily and wait until the market stabilizes to avoid losing much money, in case after reaching the stoploss the market immediately goes up. Do you think it has some potential?

Thanks in advance!

**Comment by xmatthias** [comment]
Personally, i'm not a huge fan of this for multiple reasons. 

It's got the potential to heavily spam users (assume the pair flicks between 0.44 and 0.46 over a period, because this happens to be support / resistance ...).

To be consistent, the bot would have to ask you every time, allowing you to decide every time.

I do see stoploss as "hard stop" - if your initial stop is 10% - then your capital at risk is 10(ish)% - not 20% because you "answered with yes" to a question.

You'll also create "non-reproducible" situations - which will make the backtesting / dry / live gap a LOT wider - and if you look at your results 2 weeks later, you'll have no idea why the stoploss was ignored or wasn't.

**Comment by jej** [comment]
maybe this should be considered as an independent feature, with own configuration params per pair etc. "alarms", in one word!

**Comment by MrHumanRebel** [comment]
In my opinion this should be definetelly an optional feature in the future if you may include it in a dev version. If someone does not want to be spammed by a stoploss warning, i definetelly agree on that. But if you could limit the amount of stoploss warning messages of one trade to a maximum amount for example one stoploss alert/trade, then spamming should not be a problem at all, if someone wants to know when a current trade is about to reach the given stoploss. But if you think this is completely 100% pointless then we have nothing to talk about, it was just an idea i wanted to share you guys. Thanks in advance!

**Comment by xmatthias** [comment]
as said above, the "human in the loop" you're proposing is not something that i see as fitting for freqtrade.

it can lead to many strange situations
* user didn't respond  ... do we then sell with the initial stop? or wait for the user to maybe respond (putting 100% of that trade at risk)?
* user responded with "don't sell" - but causes a huge loss because of that
* bot only waits for X seconds to let the user respond to avoid point 1 ... -> user responded "just after" the possible window - now the user wish is not respected as he was late by 2 seconds
...

In the end, it'll make results uncontrollable, and will spam you with unnecessary notifications and put stress on the user, while running a bot is supposed to Automate trading for the user.

If you only get asked once per trade it would be rather pointless, assuming a pair bounces around in the "decision" boundary - you tell the bot "don't sell" - it recovers, and a day later falls to the same area again - so agian we won't sell -> causing potential high loss should markets not recover.


In summary - i see the risk for this way too high to be usefull to most users who simply want to automate their trading.
